openedx/core/djangoapps/credit/api.py

```diff
@@ -111,6 +111,93 @@ def get_credit_requirements(course_key, namespace=None):
     ]
 
 
+def get_credit_requirement_status(course_key, username):
+    """ Retrieve the user's status for each credit requirement in the course.
+
+    Args:
+        course_key (CourseKey): The identifier for course
+        username (str): The identifier of the user
+
+    Example:
+        >>> get_credit_requirement_status("course-v1-edX-DemoX-1T2015", "john")
+
+                [
+                    {
+                        "namespace": "verification",
+                        "name": "verification",
+                        "criteria": {},
+                        "status": "satisfied",
+                    },
+                    {
+                        "namespace": "reverification",
+                        "name": "midterm",
```

Comment:
the `name` field here should have `location` of the module in case of `xblock` e.g., `i4x://edX/DemoX/edx-reverification-block/assessment_uuid"`